mne/morph.py

```diff
@@ -0,0 +1,998 @@
+        The default is 'fsaverage'.
+    data_from : array | list of two arrays | str | None
+        If morphing a surface source estimate, data_from is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_from is the absolute path
+        where subject_from/mri/brain.mgz is stored
+        If None data_from will be found through subject_from if present
+    data_to : array | list of two arrays | str | None
+        If morphing a surface source estimate , data_to is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_to is the absolute path
+        where subject_to/mri/brain.mgz is stored
+        If None data_to will be found through subject_to if present
+    subjects_dir : str, or None
+        Path to SUBJECTS_DIR if it is not set in the environment.
+    **options : **kwargs
+        Keyword arguments for the respective morphing function. For surface
+        source estimates see :func:`mne.compute_morph_matrix` and for volume
+        source estimates :func:`mne.compute_morph_sdr`
+
+    Attributes
+    ----------
+    type : str | None
+        Type of source estimate. E.g. 'vol' or 'surf'
+    subjects_dir : str
+        The path to the FreeSurfer subjects reconstructions.
+        It corresponds to FreeSurfer environment variable SUBJECTS_DIR.
+    subject_from : str
+        Name of the subject from which to morph as named in the SUBJECTS_DIR
+    subject_to : str
+        Name of the subject to which to morph as named in the SUBJECTS_DIR
+    vol_info : dict
+        Information about reference volumes for volumetric morph. Additional
+        information required to transform volume source estimate into
+        Nifti1Image using as_volume
+    src_info : dict
+        Information required to transform volume source estimate into
+        Nifti1Image using as_volume
+    morph_data : Trans | dict
+        If morphing a surface source estimate , morph_data is an instance
+        of Trans to morph the data
+        If morphing a volume source estimate morph_data is a dict storing
+        linear and non-linear mapping information.
+    options : **kwargs
+        Keyword arguments for the respective morphing function. For surface
+        source estimates see :func:`mne.compute_morph_matrix` and for volume
+        source estimates :func:`mne.compute_morph_sdr`
+
+    Notes
+    -----
+    .. versionadded:: X.X.X
+
+    See Also
+    --------
+
+    """
+
+    # initialize and pre-compute morph
+    def __init__(self, src, subject_from=None, subject_to='fsaverage',
+                 data_from=None, data_to=None, subjects_dir=None, **options):
+        self.type = None
+
+        if isinstance(src, string_types):
+            from mne import read_forward_solution
+            try:
+                src = read_forward_solution(src)['src']
+            except IOError:
+                IOError('cannot locate file: ' + src)
```

Comment:
it's the responsability of read_forward_solution to raise the error. Don't do it here.